diff --git a/CMakeLists.txt b/CMakeLists.txt
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -1,4 +1,4 @@
-cmake_minimum_required(VERSION 3.10)
+cmake_minimum_required(VERSION 3.20)
 project(nativefiledialog-extended VERSION 1.3.0)
 
 set(nfd_ROOT_PROJECT OFF)
diff --git a/src/CMakeLists.txt b/src/CMakeLists.txt
--- a/src/CMakeLists.txt
+++ b/src/CMakeLists.txt
@@ -1,10 +1,10 @@
 set(TARGET_NAME nfd)
 
 set(PUBLIC_HEADER_FILES
-  include/nfd.h
-  include/nfd.hpp
-  include/nfd_sdl2.h
-  include/nfd_glfw3.h)
+  include/nfd/nfd.h
+  include/nfd/nfd.hpp
+  include/nfd/nfd_sdl2.h
+  include/nfd/nfd_glfw3.h)
 
 set(SOURCE_FILES ${PUBLIC_HEADER_FILES})
 
diff --git a/src/include/nfd.h b/src/include/nfd/nfd.h
similarity index 100%
rename from src/include/nfd.h
rename to src/include/nfd/nfd.h
diff --git a/src/include/nfd.hpp b/src/include/nfd/nfd.hpp
similarity index 99%
rename from src/include/nfd.hpp
rename to src/include/nfd/nfd.hpp
--- a/src/include/nfd.hpp
+++ b/src/include/nfd/nfd.hpp
@@ -13,7 +13,7 @@
 #ifndef _NFD_HPP
 #define _NFD_HPP
 
-#include <nfd.h>
+#include <nfd/nfd.h>
 #include <cstddef>  // for std::size_t
 #include <memory>   // for std::unique_ptr
 #ifdef NFD_THROWS_EXCEPTIONS
diff --git a/src/include/nfd_glfw3.h b/src/include/nfd/nfd_glfw3.h
similarity index 100%
rename from src/include/nfd_glfw3.h
rename to src/include/nfd/nfd_glfw3.h
diff --git a/src/include/nfd_sdl2.h b/src/include/nfd/nfd_sdl2.h
similarity index 100%
rename from src/include/nfd_sdl2.h
rename to src/include/nfd/nfd_sdl2.h
diff --git a/src/nfd_cocoa.m b/src/nfd_cocoa.m
--- a/src/nfd_cocoa.m
+++ b/src/nfd_cocoa.m
@@ -7,7 +7,7 @@
 
 #include <AppKit/AppKit.h>
 #include <Availability.h>
-#include "nfd.h"
+#include "nfd/nfd.h"
 
 // MacOS is deprecating the allowedFileTypes property in favour of allowedContentTypes, so we have
 // to introduce this breaking change.  Define NFD_MACOS_ALLOWEDCONTENTTYPES to 1 to have it set the
diff --git a/src/nfd_gtk.cpp b/src/nfd_gtk.cpp
--- a/src/nfd_gtk.cpp
+++ b/src/nfd_gtk.cpp
@@ -16,7 +16,7 @@
 #include <stdlib.h>
 #include <string.h>
 
-#include "nfd.h"
+#include "nfd/nfd.h"
 
 /*
 Define NFD_CASE_SENSITIVE_FILTER if you want file filters to be case-sensitive.  The default
diff --git a/src/nfd_portal.cpp b/src/nfd_portal.cpp
--- a/src/nfd_portal.cpp
+++ b/src/nfd_portal.cpp
@@ -26,7 +26,7 @@
 #define getrandom(buf, sz, flags) syscall(SYS_getrandom, buf, sz, flags)
 #endif
 
-#include "nfd.h"
+#include "nfd/nfd.h"
 
 /*
 Define NFD_APPEND_EXTENSION if you want the file extension to be appended when missing. Linux
diff --git a/src/nfd_win.cpp b/src/nfd_win.cpp
--- a/src/nfd_win.cpp
+++ b/src/nfd_win.cpp
@@ -28,7 +28,7 @@ struct IUnknown;  // Workaround for "combaseapi.h(229): error C2187: syntax erro
 #include <stdio.h>
 #include <wchar.h>
 #include <windows.h>
-#include "nfd.h"
+#include "nfd/nfd.h"
 
 namespace {
 
